ketcher-autotests/tests/API/view-only-mode.spec.ts

```diff
@@ -0,0 +1,544 @@
+    page,
+  }) => {
+    /*
+    Test case: https://github.com/epam/ketcher/issues/4965
+    Description: View-only mode is still turned off after two requests in a row window.ketcher.editor.setOptions(JSON.stringify({ viewOnlyMode: false}))
+    */
+    await enableViewOnlyModeBySetOptions(page);
+    await takePageScreenshot(page);
+    await disableViewOnlyModeBySetOptions(page);
+    await disableViewOnlyModeBySetOptions(page);
+    await takePageScreenshot(page);
+  });
+
+  test('Get an error in console after sending request with the wrong parameters window.ketcher.editor.setOptions(JSON.stringify({ viewOnlyMode123: false123}))', async ({
+    page,
+  }) => {
+    /*
+    Test case: https://github.com/epam/ketcher/issues/4965
+    Description: Error in console after sending request with the wrong parameters window.ketcher.editor.setOptions(JSON.stringify({ viewOnlyMode123: false}))
+    */
+    page.on('console', (msg) => {
+      if (msg.type() === 'error') {
+        test.fail(
+          msg.type() === 'error',
+          `There is error in console: ${msg.text}`,
+        );
+      }
+    });
+    await page.evaluate(() => {
+      window.ketcher.editor.setOptions(
+        JSON.stringify({ viewOnlyMode123: `false123` }),
+      );
+    });
+  });
+
+  test('Add to canvas different elements, turn on view only mode, verify that all editing tools are disabled in toolbars for elements', async ({
+    page,
+  }) => {
+    /*
+    Test case: https://github.com/epam/ketcher/issues/4965
+    Description: All editing tools are disabled in toolbars for elements
+    */
+    await openFileAndAddToCanvasAsNewProject(
+      'KET/images-png-50-with-50-structures.ket',
+      page,
+    );
+    await enableViewOnlyModeBySetOptions(page);
+    await takePageScreenshot(page);
+  });
+
+  test('Turn on view-only mode, add to canvas different elements from file, verify that all editing tools are disabled in toolbars for elements', async ({
+    page,
+  }) => {
+    /*
+    Test case: https://github.com/epam/ketcher/issues/4965
+    Description: Turn on view-only mode, add to canvas different elements from file, verify that all editing tools are disabled in toolbars for elements
+    */
+    await enableViewOnlyModeBySetOptions(page);
+    await openFileAndAddToCanvasAsNewProject(
+      'KET/images-png-50-with-50-structures.ket',
+      page,
+    );
+    await takePageScreenshot(page);
+  });
+
+  test('Verify that the following tools and functions are enabled in view-only mode(Open, Save, Copy) ', async ({
+    page,
+  }) => {
+    /*
+    Test case: https://github.com/epam/ketcher/issues/4965
+    Description: Tools and functions are enabled in view-only mode(Open, Save, Copy) 
+    */
+    await selectRingButton(RingButton.Benzene, page);
+    await clickInTheMiddleOfTheScreen(page);
+    await enableViewOnlyModeBySetOptions(page);
+    await page.keyboard.press('Control+a');
```

Comment:
Better to use selectAllStructuresOnCanvas(page: Page) since canvas selection is asynchronous function and could lasts long